src/JoinMonster/ConnectionUtils.cs

```diff
@@ -27,8 +28,19 @@ public static string ObjectToCursor(object obj)
 
         public static IDictionary<string, object> CursorToObject(string cursor)
         {
-            var str = Encoding.UTF8.GetString(Convert.FromBase64String(cursor));
-            return JsonSerializer.Deserialize<IDictionary<string, object>>(str);
+            try
+            {
+                var str = Encoding.UTF8.GetString(Convert.FromBase64String(cursor));
+                return JsonSerializer.Deserialize<IDictionary<string, object>>(str);
+            }
+            catch (FormatException)
```

Comment:
```suggestion
catch (Exception ex) when (ex is FormatException || ex is JsonException)
{
```
we could also use this fancy syntax right ? If we'd like it

Reply:
Updated with an even fancier syntax :)